torch/_decomp/decompositions.py

```diff
@@ -1776,10 +1780,11 @@ def index_add_(
     alpha: NumberType = 1,
 ):
     dim = utils.canonicalize_dims(x.ndim, dim)
-    utils.check(
-        index.ndim <= 1,
-        lambda: f"Index should have dimension 1 or 0 (got {index.ndim})",
-    )
+    # Fails incorrectly for bool tensor
```

Comment:
What do you mean?

Reply:
@albanD This is the same issue as commented out check on like 909. `addmm(bool_tensor1, bool_tensor2, bool_tensor3, beta=1, alpha=1)` fails this check, but works fine in eager mode.

Do you know why this check was added? Are there some checks in C++ that we're trying (incorrectly) to emulate?

Reply:
I am not sure tbh. But I would expect that yes. I'm sure Natalia would know.

Reply:
Why do we even support addmm for bools? I'm pretty sure we don't support it on cuda.

Reply:
This is the check, I believe:
https://github.com/pytorch/pytorch/blob/f903f1ab343fea72177f29fc8d453febcaad8905/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L284-L285
As to why does it fails for bools... I don't know.

I am currently putting together a more thorough test for decompositions. This test that has uncovered a few issues in some operations, in particular for bools. No idea what's going on there just yet. I plan to fix all the issues the test has uncovered before putting up a PR. Hopefully that'll be today or tomorrow. For now, I'm leaving here this fantastic behaviour it uncovered:

```python
import torch msk = torch.tensor([False])
b = torch.tensor([False])
print(torch.where(msk, True, b).dtype)
print(torch.ops.aten.where.ScalarSelf(msk, True, b).dtype)
print(torch._refs.where(msk, True, b).dtype)
```
outputs
```
torch.bool
torch.int64
torch.bool
```
This happens just for boolean tensors, all the other types work just fine.